ui/src/features/common/promotion-directive-step-status/utils.ts

```diff
@@ -0,0 +1,29 @@
+import { PromotionStatusPhase } from '@ui/features/common/promotion-status/utils';
+import { PromotionStatus } from '@ui/gen/v1alpha1/generated_pb';
+
+// UI concludes from Promotion's status data
+export enum PromotionDirectiveStepStatus {
+  RUNNING,
+  FAILED,
+  SUCCESS
+}
+
+export const getPromotionDirectiveStepStatus = (
```

Comment:
@krancour / @hiddeco can you please read this logic if anything is inaccurate?

Basically for any step ran, UI tries to decide what is state of all steps